docs/components/query-frontend.md

```diff
@@ -44,22 +44,55 @@ Query Frontend supports a retry mechanism to retry query when HTTP requests are
 ### Caching
 
 Query Frontend supports caching query results and reuses them on subsequent queries. If the cached results are incomplete,
-Query Frontend calculates the required subqueries and executes them in parallel on downstream queriers. Query Frontend can optionally align queries with their step parameter to improve the cacheability of the query results.
+Query Frontend calculates the required subqueries and executes them in parallel on downstream queriers.
+Query Frontend can optionally align queries with their step parameter to improve the cacheability of the query results.
+It uses the cortex cache module so supports all that is supported there.
+For more details about each configuration parameter visit the [cortex documentation](https://cortexmetrics.io/docs/configuration/configuration-file/).
 
-Currently, only in-memory cache (fifo cache) is supported. An example config:
-
-[embedmd]:# (../flags/config_response_cache_in_memory.txt yaml)
+[embedmd]:# (../flags/config_frontend_cache.txt yaml)
 ```yaml
-type: IN-MEMORY
-config:
-  max_size: ""
-  max_size_items: 0
-  validity: 0s
+- enable_fifocache: false
```

Comment:
These configuration format kind of bothers me. Is this really what we want to do? 
This configuration format not idiomatic to Thanos. And also we are just exposing a configuration API from a dependency of ours directly. 
Can we do something about this?

Reply:
changing this means more maintenance, I agree that it doesn't look nice, but I think we should work towards changing this upstream in Thanos. keeping it tied together with Cortex means that people using the cortex front end can switch with their existing configs.

Reply:
Does Cortex use similar configuration format as Thanos does? Can we make it happen? Otherwise I'm willing to take the responsibility of maintenance burden in order to keep Thanos configuration API consistent.

Reply:
This is the first step to make things more consistent, and my idea is to slowly contribute upstream to make things better. Also, the next step would be using the same code for the memcache client - at the moment Cortex has its own and Thanos uses a slightly different version.

Btw why do you think this change makes the API inconsistent?

Reply:
The previous style where we define type and specify the properties what we use across the board. Existence of a config enables corresponding backend. I find it more idiomatic for Thanos. e.g https://thanos.io/v0.15/components/store/

Reply:
This config is completely different from the Thanos way. Cortex supports configuring multiple cache backend to be a tiered cache, while Thanos can only configure one.

I think it is hard to convert this config to some Thanos format

Reply:
yeah I completely forgot about that -  another reason to keep it with the cortex configs logic.

Reply:
Interesting. How one would configure tiered cache in Cortex then even? :thinking: 

I would aim for our config type though - even if it reproduces a bit of configuration. We would rather aim for 1:1 configuration to other caching usages (index, bucket on store), then aiming for Cortex configuration style I think. The good question to be answered, is how the tiered cache if we want this would look like. I think I would see this as multiple items in this file with different types so all find on our side.

No need to force Cortex clients to have same config as well - for this we would need bigger discussion and maintaining just static configuration structs for better discoverability, readability sounds like not a bad idea to me (: But there is some work yes - let's do here whatever we can (e.g just Memcached) and produce issues to add rest.

WDYT? @krasi-georgiev 

And thanks for this work!